src/Modules/Settings/Dnn.PersonaBar.SiteSettings/Services/SiteSettingsController.cs

```diff
@@ -804,6 +805,12 @@ public HttpResponseMessage AddProfileProperty(UpdateProfilePropertyRequest reque
 
                 if (ValidateProperty(property))
                 {
+                    Regex propertyNameRegex = new Regex("^[a-zA-Z0-9]+$");
+                    if (!propertyNameRegex.Match(property.PropertyName).Success)
+                    {
+                        return Request.CreateErrorResponse(HttpStatusCode.BadRequest, string.Format(Localization.GetString("NoSpecialCharacterName.Text", Components.Constants.Constants.LocalResourcesFile)));
+                    }
```

Comment:
I think this belongs inside `ValidateProperty`, especially that it's being called in other actions too